app/lib/search/mem_index.dart

```diff
@@ -581,3 +588,48 @@ class _PackageNameIndex {
     return Score(values);
   }
 }
+
+class _LikeScore {
+  final String package;
+  int likeCount = 0;
+  double score = 0.0;
+
+  _LikeScore(this.package, {this.likeCount = 0, this.score = 0.0});
+}
+
+class _LikeTracker {
+  final _values = <String, _LikeScore>{};
+  bool _changed = false;
+
+  double getLikeScore(String package) {
+    if (_changed) {
+      _updateScores();
+    }
+    return _values[package]?.score ?? 0.0;
+  }
+
+  void trackLikeCount(String package, int likeCount) {
+    final v = _values.putIfAbsent(package, () => _LikeScore(package));
+    if (v.likeCount != likeCount) {
+      _changed = true;
+      v.likeCount = likeCount;
+    }
+  }
+
+  void removePackage(String package) {
+    final removed = _values.remove(package);
+    _changed |= removed != null;
+  }
+
+  void _updateScores() {
```

Comment:
One line doc-comment explaining what this does wouldn't work... it updates `score` to be a ranking from 1 to 0.